components/director/internal/domain/application/resolver.go

```diff
@@ -40,40 +58,150 @@ func NewResolver(svc *service, apiSvc APIService, eventAPISvc EventAPIService, d
 }
 
 func (r *Resolver) Applications(ctx context.Context, filter []*graphql.LabelFilter, first *int, after *graphql.PageCursor) (*graphql.ApplicationPage, error) {
+	labelFilter := labelfilter.MultipleFromGraphQL(filter)
+
+	var cursor string
+	if after != nil {
+		cursor = string(*after)
+	}
+
+	appPage, err := r.svc.List(ctx, labelFilter, first, &cursor)
+	if err != nil {
+		return nil, err
+	}
+
+	gqlApps := r.converter.MultipleToGraphQL(appPage.Data)
+	totalCount := len(gqlApps)
+
 	return &graphql.ApplicationPage{
-		Data:       []*graphql.Application{},
-		TotalCount: 0,
+		Data:       gqlApps,
+		TotalCount: totalCount,
 		PageInfo: &graphql.PageInfo{
-			HasNextPage: false,
+			StartCursor: graphql.PageCursor(appPage.PageInfo.StartCursor),
+			EndCursor:   graphql.PageCursor(appPage.PageInfo.EndCursor),
+			HasNextPage: appPage.PageInfo.HasNextPage,
 		},
 	}, nil
 }
 
 func (r *Resolver) Application(ctx context.Context, id string) (*graphql.Application, error) {
-	panic("not implemented")
+	app, err := r.svc.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	return r.converter.ToGraphQL(app), nil
 }
 
 func (r *Resolver) CreateApplication(ctx context.Context, in graphql.ApplicationInput) (*graphql.Application, error) {
-	panic("not implemented")
+	convertedIn := r.converter.InputFromGraphQL(in)
+
+	id, err := r.svc.Create(ctx, convertedIn)
+	if err != nil {
+		return nil, err
+	}
+
+	app, err := r.svc.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	gqlApp := r.converter.ToGraphQL(app)
+
+	return gqlApp, nil
 }
 func (r *Resolver) UpdateApplication(ctx context.Context, id string, in graphql.ApplicationInput) (*graphql.Application, error) {
-	panic("not implemented")
+	convertedIn := r.converter.InputFromGraphQL(in)
+
+	err := r.svc.Update(ctx, id, convertedIn)
+	if err != nil {
+		return nil, err
+	}
+
+	app, err := r.svc.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	gqlApp := r.converter.ToGraphQL(app)
+
+	return gqlApp, nil
 }
 func (r *Resolver) DeleteApplication(ctx context.Context, id string) (*graphql.Application, error) {
-	panic("not implemented")
+	app, err := r.svc.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	deletedApp := r.converter.ToGraphQL(app)
+
+	err = r.svc.Delete(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	return deletedApp, nil
 }
-func (r *Resolver) AddApplicationLabel(ctx context.Context, applicationID string, label string, values []string) ([]string, error) {
-	panic("not implemented")
+func (r *Resolver) AddApplicationLabel(ctx context.Context, applicationID string, key string, values []string) (*graphql.Label, error) {
+	err := r.svc.AddLabel(ctx, applicationID, key, values)
```

Comment:
As you know, Labels/Annotations can be extended soon. If this happens, we will have to change signatures for our services. Please create proper structs for those 2 types and pass it as an argument for the services, instead of passing `key` and `values`.

Reply:
Okay, I will create a follow-up task